File: 백준/Gold/1068. 트리/트리.cc
#include <bits/stdc++.h>
using namespace std;

int n, m, root, ret;
vector<int> adj[50];

void go(int u) {
  if (u == m) return;
  int cnt = 0;
  for (int v : adj[u]) {
    if (v == m) continue;
    go(v);
    cnt++;
  }

  if (cnt == 0) ret++;
}

int main() {
  cin >> n;
  for (int i = 0; i < n; i++) {
    int p;
    cin >> p;

    if (p == -1) {
      root = i;
    } else {
      adj[p].push_back(i);
    }
  }
  cin >> m;

  go(root);

  cout << ret;

  return 0;
}

File: 백준/Gold/1068. 트리/트리.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : './input.txt').toString().trim().split('\n');

const N = Number(input[0]);
const parents = input[1].split(' ').map(Number);
const M = Number(input[2]);

const adj = Array.from({length: N}, () => []);
let root;

parents.forEach((node, i) => {
    if(node === -1) root = i;
    else adj[node].push(i);
})

const go = (here) => {
    let cnt = 0, ret = 0;

    for(const there of adj[here]){
        if(M === there) continue;

        cnt ++;
        ret += go(there);
    }
    
    if(cnt === 0) return 1;

    return ret;
}

const answer = () => {
    if(M == root){
        console.log(0);
    }else{
        console.log(go(root));
    }
    
}

answer();